include/seqan3/alignment/matrix/alignment_optimum.hpp

```diff
@@ -0,0 +1,86 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_optimum.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include <seqan3/alignment/matrix/alignment_coordinate.hpp>
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+#include <seqan3/std/concepts>
+
+namespace seqan3::detail
+{
+
+/*!\brief Stores the current optimum of the alignment algorithms.
+ * \ingroup alignment_matrix
+ * \tparam score_t The type of the tracked alignment score.
+ *
+ * \details
+ *
+ * This is an aggregate type, so the score needs to be passed before the seqan3::alignment_coordinate during
+ * construction.
+ */
+template <Arithmetic score_t>
+struct alignment_optimum
+{
+    //!\brief The optimal score.
+    score_t score{std::numeric_limits<score_t>::lowest()};
+    //!\brief The corresponding coordinate within the alignment matrix.
+    alignment_coordinate coordinate{};
+};
+
+/*!\name Type deduction guides
+ * \{
+ */
+//!\brief Default constructed objects deduce to `int32_t`.
+alignment_optimum() -> alignment_optimum<int32_t>;
+
+template <Arithmetic score_t>
+alignment_optimum(score_t const, alignment_coordinate const) ->
+    alignment_optimum<std::remove_reference_t<score_t>>;
+//!\}
+
+/*!\brief A less than comparator for two seqan3::detail::alignment_optimum objects.
+ * \ingroup alignment_matrix
+ *
+ * \details
+ *
+ * This function object is used in std::max functions to compare two seqan3::detail::alignment_optimum objects.
```

Comment:
It is not a lambda, because it would be the same and it is a better style if lambdas are used within scoped contexts and not declared globally. Second, I might overload it for different scoring types. Keeping it encapsulated in a class (possibly class template) seems to be the right way to go.